lib/cylc/task_proxy.py

```diff
@@ -1218,6 +1424,16 @@ def process_incoming_messages(self):
             except Queue.Empty:
                 break
             queue.task_done()
+        # Retry any failed event handlers
+        for key, try_state in self.event_handler_try_states.items():
+            if try_state.ctx and try_state.is_delay_done():
+                ctx = SuiteProcContext(
+                    try_state.ctx.cmd_type,
+                    try_state.ctx.cmd,
+                    **try_state.ctx.cmd_kwargs)
+                try_state.ctx = ctx
+                SuiteProcPool.get_inst().put_command(
+                    ctx, self.event_handler_callback)
 
```

Comment:
For initial convenience. It is the most obvious place without creating a new function, given that most event handlers originate from incoming messages. I'll tidy this up and move the logic to its own function.